File: 00-quick-start/code/sample.js
// sample.js - A simple user service with some issues
import mysql from 'mysql';

function getUser(userId) {
  const query = "SELECT * FROM users WHERE id = " + userId;
  return db.query(query);
}

function updateProfile(req, res) {
  const name = req.body.name;
  res.send("<h1>Welcome " + name + "</h1>");
}

function login(username, password) {
  if (password == "admin123") {
    return { token: "abc123", admin: true };
  }
}

export {
  getUser,
  updateProfile,
  login
};

File: 00-quick-start/code/sample.py
# sample.py - A simple user service with some issues
import sqlite3
from flask import Flask, request

app = Flask(__name__)

def get_user(user_id):
    conn = sqlite3.connect('users.db')
    query = "SELECT * FROM users WHERE id = " + user_id
    return conn.execute(query).fetchone()

@app.route('/profile', methods=['POST'])
def update_profile():
    name = request.form['name']
    return f"<h1>Welcome {name}</h1>"

def login(username, password):
    if password == "admin123":
        return {"token": "abc123", "admin": True}
